src/services/workspace-client/devWorkspaceClient.ts

```diff
@@ -0,0 +1,193 @@
+  private initializing: Promise<void>;
+
+  constructor(@inject(KeycloakSetupService) keycloakSetupService: KeycloakSetupService) {
+    super(keycloakSetupService);
+    this.axios.defaults.baseURL = '/api/unsupported/k8s';
+    this.client = DevWorkspaceClientLibrary.getRestApi(this.axios);
+    this.workspaceApi = this.client.workspaceApi;
+    this.previousItems = new Map();
+    this.maxStatusAttempts = 10;
+  }
+
+  isEnabled(): Promise<boolean> {
+    return this.client.isDevWorkspaceApiEnabled();
+  }
+
+  async getAllWorkspaces(defaultNamespace: string): Promise<che.Workspace[]> {
+    await this.initializing;
+    const workspaces = await this.workspaceApi.listInNamespace(defaultNamespace);
+    const availableWorkspaces: che.Workspace[] = [];
+    for (const workspace of workspaces) {
+      if (!isDeleting(workspace) && !isWebTerminal(workspace)) {
+        availableWorkspaces.push(convertDevWorkspaceV2ToV1(workspace));
+      }
+    }
+    return availableWorkspaces;
+  }
+
+  async getWorkspaceByName(namespace: string, workspaceName: string): Promise<che.Workspace> {
+    let workspace = await this.workspaceApi.getByName(namespace, workspaceName);
+    let attempted = 0;
+    while ((!workspace.status || !workspace.status.phase || !workspace.status.ideUrl) && attempted < this.maxStatusAttempts) {
+      workspace = await this.workspaceApi.getByName(namespace, workspaceName);
+      this.checkForDevWorkspaceError(workspace);
+      attempted += 1;
+      await delay();
+    }
+    this.checkForDevWorkspaceError(workspace);
+    if (!workspace.status || !workspace.status.phase || !workspace.status.ideUrl) {
+      throw new Error(`Could not retrieve devworkspace status information from ${workspaceName} in namespace ${namespace}`);
+    }
+    return convertDevWorkspaceV2ToV1(workspace);
+  }
+
+  async create(devfile: IDevWorkspaceDevfile): Promise<che.Workspace> {
+    const createdWorkspace = await this.workspaceApi.create(devfile, this._defaultEditor, this._defaultPlugins);
+    return convertDevWorkspaceV2ToV1(createdWorkspace);
+  }
+
+  delete(namespace: string, name: string): void {
+    this.workspaceApi.delete(namespace, name);
+  }
+
+  async changeWorkspaceStatus(namespace: string, name: string, started: boolean): Promise<che.Workspace> {
+    const changedWorkspace = await this.workspaceApi.changeStatus(namespace, name, started);
+    this.checkForDevWorkspaceError(changedWorkspace);
+    return convertDevWorkspaceV2ToV1(changedWorkspace);
+  }
+
+  /**
+   * Initialize the given namespace
+   * @param namespace The namespace you want to initialize
+   * @returns If the namespace has been initialized
+   */
+  async initializeNamespace(namespace: string): Promise<boolean> {
+    try {
+      this.initializing = new Promise((resolve, reject) => {
+        this.workspaceApi.initializeNamespace(namespace).then(_ => {
+          resolve(undefined);
+        });
+      });
+      await this.initializing;
```

Comment:
If I understand correctly this will work in the same way

```suggestion
      await this.workspaceApi.initializeNamespace(namespace);
```